server/src/main/java/org/opensearch/index/IndexSettings.java

```diff
@@ -1024,6 +1024,9 @@ public IndexSettings(final IndexMetadata indexMetadata, final Settings nodeSetti
     }
 
     private void setSearchIdleAfter(TimeValue searchIdleAfter) {
+        if (this.isRemoteStoreEnabled) {
+            logger.warn("Search idle is not supported for remote backed indices");
+        }
```

Comment:
What is the side effect if we throw unsupported exception

Reply:
The update index setting api caller would see an error response. Sample exception when we throw `throw new IllegalArgumentException("test message")` -
```
{
  "error" : {
    "root_cause" : [
      {
        "type" : "illegal_argument_exception",
        "reason" : "test message"
      }
    ],
    "type" : "illegal_argument_exception",
    "reason" : "test message"
  },
  "status" : 400
}

```

Reply:
It's better to fail rather than silently failing to honour it.

Reply:
This requires a bit more thought, I have created https://github.com/opensearch-project/OpenSearch/issues/10787 for follow up on this.